CONTRIBUTING.md

```diff
@@ -0,0 +1,234 @@
+
+# Stacks Project Guidelines
+
+- [Introduction](#introduction)
+- [Get Started](#get-started)
+  - [System Requirements](#system-requirements)
+  - [Setup](#setup)
+- [Naming Conventions and Coding Style](#naming-conventions-and-coding-style)
+- [Creating a New Plugin](#creating-a-new-plugin)
+  - [Generator Analysis](#generator-analysis)
+  - [Development process](#development-process)
+- [Testing](#testing)
+  - [Unit Testing](#unit-testing)
+  - [End to end testing](#end-to-end-testing)
+  - [Testing packages locally](#testing-packages-locally)
+- [Contributing Changes](#contributing-changes)
+- [Releasing packages and publishing to NPM](#releasing-packages-and-publishing-to-npm)
+
+## Introduction
+
+Welcome to the Stacks project! This document provides guidelines to help you get started with the Stacks workspace and develop new plugins or contribute to existing ones. It covers setup instructions, development practices, testing, and contribution guidelines.
+
+## Get Started
+
+### System Requirements
+
+To set up the Stacks workspace, ensure that you have the following installed on your system:
+
+- npm
+
+- Node.js version 18 or above
+
+The following are recommended plugins for visual studio code:

```

Comment:
Are there any notes for the Webstorm IDE? If there isn't much around the repo I can ask Cerden and George and see that we recommend as we all use Webstorm.

Reply:
Agreed to complete at a later date